### DISCUSSION TITLE
Optimization on NetworkTransform

### DISCUSSION THREAD

**Discussion by Ozden79** [FirstGearGames/FishNet#553] [Feature Request]
Hi There,

I would like to propose a feature addition to NetworkTransform object. Normally this object has been optimized for changes on transform values which is great but in certain conditions (likely AI controlled objects) the transform value changes are predictable and a small behavior change in this object might eliminate some good network traffic.

I would like to describe it with a simple scenario for position property. Suppose that in a war themed game, an enemy is controlled by AI where it's directed to go from position A to position B in X seconds. Currently all these positions are transferred to the clients with each update. If there can be a flag which defines the position change as direction per second on the NetworkTransform, this way the client can update the position on client side until it receives another message which removes this flag so the positioning can be again absolute as it's normally.

You may tell that you can do this by adding client side methods to do those on your objects but there are atleast 2 things to consider :

1) It makes the client vulnerable and cheaters can get/predict the object move ending position on the client.
2) You need to write client side extension methods to your AI code to handle this and also need to pause the NetworkTransform synchronization which would complicate things further.

Apologies if there is any area which is not clear as English is not my native language and don't hasitate to ask if you need further information.

Thanks,

Özden Irmak

**Comment by FirstGearGames** [comment]
There are already plans for something similar on the road map. 

PS: the road map became available after this post.
